### PR TITLE
op-node: fix Ecotone beacon-block-root contract deployment

### PR DESCRIPTION
**Pull request by protolambda**
**Description**

`common.Hex2Bytes` is a horrific footgun.
The upgrade-tx had no data, and the tests were comparing empty data to empty data.

This fixes the upgrade tx and EIP-4788 contract code definition.

Also fixes the action-test block-building: the beacon-roots processing isn't part of the regular state-transition of geth, and was causing issues there. Our engine-API implementation (also used in fault-proofs) needs to handle this block building behavior.

**Tests**

- Test that the deployment really puts the code in-place
- Test the same thing with code-hash check
- Test that we set the beacon-block root (and deal with activation-block edge-case)
- Test that we configure the upgrade tx correctly with non-empty input
- Test that the constants are not empty, paranoid but proven necessary
- Test that the beacon-block-roots storage is changing with new beacon block roots 

**Additional context**

This went wrong on an internal devnet. The devnet was about to be retired anyways, but thus won't have beacon-block-roots.

### CODE REVIEW

File: op-e2e/actions/ecotone_fork_test.go [hunk 1]
```diff
@@ -132,14 +134,49 @@ func TestEcotoneNetworkUpgradeTransactions(gt *testing.T) {
...
-	code, err := engine.EthClient().CodeAt(context.Background(), expected4788Address, latestBlock.Number())
-	require.NoError(t, err)
-	require.Equal(t, code, predeploys.EIP4788ContractCode)
+	require.Equal(t, predeploys.EIP4788ContractAddr, expected4788Address)
+	code, err := ethCl.CodeAt(context.Background(), expected4788Address, latestBlock.Number())
+	require.NoError(t, err)
+	require.NotEmpty(t, code)
+	require.Equal(t, predeploys.EIP4788ContractCode, code)
+	require.NotEmpty(t, predeploys.EIP4788ContractAddr)
+	require.Equal(t, predeploys.EIP4788ContractCodeHash, crypto.Keccak256Hash(code))
```
Comment (trianglesphere) [inline op-e2e/actions/ecotone_fork_test.go]: Why not a `verfiyCodeHashMatches` here?

**Review by trianglesphere — COMMENTED** [review]
gnarly bug. TY for tracking it down

File: op-e2e/actions/ecotone_fork_test.go (hunk 1 above)
Comment (sebastianst) [reply op-e2e/actions/ecotone_fork_test.go]: I guess because `verifyCodeHashMatches` doesn't assert `NotEmpty`, and doesn't assert code equality directly. I think we could actually add the `NotEmpty` assertions to `verfiyCodeHashMatches` and then use it here. Or do we also use it for empty code? In that case we could add a stricter `verifyNonEmptyCode`.

Also wondering why we pass a byte slice instead of a `common.Hash` to `verifyCodeHashMatches`? The latter would be cleaner.

File: op-e2e/actions/ecotone_fork_test.go [hunk 2]
```diff
@@ -132,14 +134,49 @@ func TestEcotoneNetworkUpgradeTransactions(gt *testing.T) {
...
+		timeIdx := common.BigToHash(new(big.Int).SetUint64(timestamp % historyBufferLength))
+
+		rootValue, err := ethCl.StorageAt(context.Background(), predeploys.EIP4788ContractAddr, rootIdx, nil)
+		require.NoError(t, err)
+		require.Equal(t, expectedHash, common.BytesToHash(rootValue), msg)
+
+		timeValue, err := ethCl.StorageAt(context.Background(), predeploys.EIP4788ContractAddr, timeIdx, nil)
+		require.NoError(t, err)
+		require.True(t, common.BytesToHash(timeValue).Big().IsUint64())
+		require.Equal(t, expectedTime, common.BytesToHash(timeValue).Big().Uint64(), msg)
```
Comment (sebastianst) [inline op-e2e/actions/ecotone_fork_test.go]: why not
```suggestion
		timeBig := new(big.Int).SetBytes(timeValue)
		require.True(t, timeBig.IsUint64())
		require.Equal(t, expectedTime, timeBig.Uint64(), msg)
```
? Might be misleading to convert a number value to a "Hash" and then back to a `big.Int`.

**Review by sebastianst — APPROVED** [review]
Thanks for fixing this and adding a lot more tests!

Two nits, otherwise lgtm.

File: op-e2e/actions/ecotone_fork_test.go (hunk 1 above)
Comment (protolambda) [reply op-e2e/actions/ecotone_fork_test.go]: I'll add the `NotEmpty` check to `verfiyCodeHashMatches`, good idea. And yes, will change it to use fixed-size type, no risk of accidental empty code hash slice then.

**Review by trianglesphere — APPROVED** [review]

**Comment by protolambda** [comment]
it got kicked from merge queue due to some git reference being failed to fetch? Failing git submodule maybe?
